### PR TITLE
pipreqs: init at 0.4.9 | pythonPackage.yarg: init at 0.1.9

### PR DESCRIPTION
**Pull request by PsyanticY**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions (ubuntu 18.04)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`) (pipreqs)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/pipreqs/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, pythonPackages }:
```
Comment (dotlambda) [inline pkgs/tools/misc/pipreqs/default.nix]: Can we use `python3Packages`?

File: pkgs/development/python-modules/yarg/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    inherit pname version;
+    sha256 = "55695bf4d1e3e7f756496c36a69ba32c40d18f821e38f61d028f6049e5e15911";
+  };  
+
+  propagatedBuildInputs = [ requests ];
+
+  # tests requires tox / requires sysctl commands
+  doCheck = false;
+
+  meta = with stdenv.lib; {
```
Comment (dotlambda) [inline pkgs/development/python-modules/yarg/default.nix]:
```suggestion
   meta = with lib; {
```

File: pkgs/development/python-modules/yarg/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, buildPythonPackage, fetchPypi, requests }:
```
Comment (dotlambda) [inline pkgs/development/python-modules/yarg/default.nix]:
```suggestion
 { lib, buildPythonPackage, fetchPypi, requests }:
```

File: pkgs/tools/misc/pipreqs/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    inherit pname version;
+    sha256 = "cec6eecc4685967b27eb386037565a737d036045f525b9eb314631a68d60e4bc";
+  };
+
+  propagatedBuildInputs = with pythonPackages; [ yarg docopt ];
+
+  # Tests requires network access. Works fine without sandboxing
+  doCheck = false;
+
+  meta = with stdenv.lib; {
```
Comment (dotlambda) [inline pkgs/tools/misc/pipreqs/default.nix]:
```suggestion
   meta = with lib; {
```

File: pkgs/tools/misc/pipreqs/default.nix (hunk 1 above)
Comment (dotlambda) [inline pkgs/tools/misc/pipreqs/default.nix]:
```suggestion
 { lib, pythonPackages }:
```

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -744,6 +744,8 @@ in {
 
   webapp2 = callPackage ../development/python-modules/webapp2 { };
 
+  yarg = callPackage ../development/python-modules/yarg { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: This needs to be moved down. I think `pyunbound` is just at the wrong place.

**Review by dotlambda — COMMENTED** [review]
Please change the first commit message to `maintainers: update psyanticy's github`

File: pkgs/tools/misc/pipreqs/default.nix (hunk 1 above)
Comment (PsyanticY) [reply pkgs/tools/misc/pipreqs/default.nix]: Yeah but seems when packaging using `python3Packages`. the binary fails for python code written in python2 (tested the binary on,`nixops` ) fails when it find `print "dsqdqs"`.

File: pkgs/top-level/python-packages.nix (hunk 5 above)
Comment (PsyanticY) [reply pkgs/top-level/python-packages.nix]: @dotlambda  true didn't see the one after `pyunbound`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/yarg/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    sha256 = "55695bf4d1e3e7f756496c36a69ba32c40d18f821e38f61d028f6049e5e15911";
+  };  
+
+  propagatedBuildInputs = [ requests ];
+
+  # tests requires tox / requires sysctl commands
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A semi hard Cornish cheese, also queries PyPI";
```
[the remark this answers is not in the document]
Comment (PsyanticY) [reply pkgs/development/python-modules/yarg/default.nix]: Thats upstream description : https://github.com/kura/yarg

File: pkgs/tools/misc/pipreqs/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{ lib, pythonPackages }:
```
[the remark this answers is not in the document]
Comment (PsyanticY) [reply pkgs/tools/misc/pipreqs/default.nix]: it fails with python3 to read source code written in python 2 (nixops for example) 
```pain@Greed:~/Documents/nixpkgs€ /nix/store/08db51gd6f2j1gzq9jcw74z6nxf5dzh9-pipreqs-0.4.9/bin/pipreqs ~/Documents/nixops/
ERROR: Failed on file: /home/pain/Documents/nixops/nix/generate-ec2-properties.py
Traceback (most recent call last):
  File "/nix/store/08db51gd6f2j1gzq9jcw74z6nxf5dzh9-pipreqs-0.4.9/bin/.pipreqs-wrapped", line 12, in <module>
    sys.exit(main())
  File "/nix/store/08db51gd6f2j1gzq9jcw74z6nxf5dzh9-pipreqs-0.4.9/lib/python3.7/site-packages/pipreqs/pipreqs.py", line 396, in main
    init(args)
  File "/nix/store/08db51gd6f2j1gzq9jcw74z6nxf5dzh9-pipreqs-0.4.9/lib/python3.7/site-packages/pipreqs/pipreqs.py", line 341, in init
    extra_ignore_dirs=extra_ignore_dirs)
  File "/nix/store/08db51gd6f2j1gzq9jcw74z6nxf5dzh9-pipreqs-0.4.9/lib/python3.7/site-packages/pipreqs/pipreqs.py", line 91, in get_all_imports
    raise exc
  File "/nix/store/08db51gd6f2j1gzq9jcw74z6nxf5dzh9-pipreqs-0.4.9/lib/python3.7/site-packages/pipreqs/pipreqs.py", line 77, in get_all_imports
    tree = ast.parse(contents)
  File "/nix/store/6lm4gi5iv8fbf1b1mm6g3gfnnv63f1gn-python3-3.7.1/lib/python3.7/ast.py", line 35, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 23
    print '{'
            ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print('{')?
```

File: pkgs/development/python-modules/yarg/default.nix [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildPythonPackage, fetchFromGitHub, requests, nose, mock }:
+
+buildPythonPackage rec {
+  pname = "yarg";
+  version = "0.1.9";
+
+  src = fetchFromGitHub  {
+    owner = "kura";
+    repo = pname;
+    rev = "${version}";
```
Comment (dotlambda) [inline pkgs/development/python-modules/yarg/default.nix]:
```suggestion
    rev = version;
```

File: pkgs/tools/misc/pipreqs/default.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/tools/misc/pipreqs/default.nix]: If I understand it correctly, the Python 2 version of pipreqs cannot be used to read Python 3 source code and vice versa. In that case, this is a package that can be used with different Python versions and has to be put into `python-packages.nix`.
Please also add a comment explaining the choice.

File: pkgs/development/python-modules/yarg/default.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/yarg/default.nix]: I know it is but in this case, I'd prefer not to follow upstream. You can take Fedora as an inspiration: https://repology.org/metapackage/python:yarg/packages.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/pipreqs/default.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (PsyanticY) [reply pkgs/tools/misc/pipreqs/default.nix]: @dotlambda It works fine on both with pipreqs packaged by python2. It fails on python2 code when packaging pipreqs with python3 .

**Comment by PsyanticY** [comment]
@dotlambda Added the comment as requested, Would you please have a look. ^^

File: pkgs/development/python-modules/yarg/default.nix (hunk 8 above)
Comment (dotlambda) [reply pkgs/development/python-modules/yarg/default.nix]: This should still be changed.

File: pkgs/tools/misc/pipreqs/default.nix [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+# Using python 2 as when packaging with python 3 pipreqs fails to parse python 2 code.
```
Comment (dotlambda) [inline pkgs/tools/misc/pipreqs/default.nix]: :+1:

File: pkgs/tools/misc/pipreqs/default.nix [hunk 10]
```diff
@@ -0,0 +1,25 @@
+{ lib, pythonPackages }:
```
Comment (dotlambda) [inline pkgs/tools/misc/pipreqs/default.nix]:
```suggestion
{ lib, python2Packages }:
```

**Comment by PsyanticY** [comment]
@dotlambda  Updated the last ones as well.

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.yarg python3.pkgs.yarg pipreqs

**Comment by PsyanticY** [comment]
@dotlambda  Why GrahamcOfBorg build logs are not shown like usual (just noticed this on couple of recent PR).

**Pull request merged**